pkgs/top-level/default.nix

```diff
@@ -19,83 +25,25 @@
                && system != "x86_64-solaris"
                && system != "x86_64-kfreebsd-gnu")
 
```

Comment:
I like that idea, but I see it less as something that should have been done already than something that's good to do in addition---I haven't messed with any parameters that actual packages receive yet.

That said, I'm fine just making it an additional commit that's part of this rather than a separate PR.